R/mobr.R

```diff
@@ -1016,197 +956,322 @@ pairwise_t = function(dat_sp, dat_plot, groups, lower_N = NA) {
   return(out)
 }
 
-plot_grp_rads = function(comm_obj, env_var, col=NA,
-                         log='') {
-  #plot group pooled rank species abundance distribution
-  par(mfrow = c(1, 1))
-  env_data = comm_obj$env[ , env_var]
-  grps = unique(env_data)
-  if (is.na(col[1])) 
-    col = rainbow(length(grps))
-  sads = aggregate(comm_obj$comm, by=list(comm_obj$env[ , env_var]), 
-                   sum)
-  grps = as.character(sads[,1])
-  sads = as.matrix(sads[,-1])
-  sads = ifelse(sads == 0, NA, sads)
-  plot(1:10, 1:10, type='n', 
-       xlab='rank', ylab='abundance',
-       log=log, xlim=c(1, ncol(comm_obj$comm)), 
-       ylim=range(sads, na.rm=T), 
-       cex.lab = 1.5, cex.axis = 1.5)
-  for(i in 1:nrow(sads)) 
-    lines(1:sum(!is.na(sads[i, ])), sort(sads[i, ], dec=T),
-          col=col[i], lwd=2)
-  legend('topright', grps, col=col, bty='n', lty=1,
-         lwd=3, cex=2)
+#' Plot distributions of species abundance
+#' 
+#' @param mob_in a 'mob_in' class object produced by 'make_mob_in'
+#' @param env_var a string that specifies the column name in mob_in$env that 
+#'   specifies the grouping variable. 
+#' @param type either 'sad' or 'rad' for species abundance vs rank abundance distribution
+#' @param pooled boolean specifying if abundances should be pooled at the group level or
+#'   not
+#' @param col optional vector of colors.
+#' @param log specify which axes to apply log transformations to.
+#' @importFrom scales alpha
+#' @export
+#' @examples
+#' data(inv_comm)
+#' data(inv_plot_attr)
+#' inv_mob_in = make_mob_in(inv_comm, inv_plot_attr)
+#' plot_abu(inv_mob_in, 'group', 'sad', pooled=F, log='x')
+#' plot_abu(inv_mob_in, 'group', 'rad', pooled=T, log='x')
+plot_abu = function(mob_in, env_var, type=c('sad', 'rad'), pooled=FALSE,
+                    col=NA, log='', leg_loc = 'topleft') {
+    env_data = mob_in$env[ , env_var]
+    grps = unique(env_data)
+    if (is.na(col[1])) 
+        col = rainbow(length(grps))
+    if ('sad' == type) {
+        plot(1, type = "n", xlab = "% abundance (log scale)", ylab = "% species", 
+             xlim = c(0.01, 1), ylim = c(0.01, 1), log = log)
+        for (i in 1:length(grps)) {
+            col_grp = col[i]
```

Comment:
Agreed - in general the functions are not very robust right now to users that aren't paying attention to what they supply our functions. If time allows adding argument checks would be a good thing to do. More importantly though we should implement some formal tests to go along with the package and to aid future development.